homeassistant/components/html5/notify.py

```diff
@@ -518,3 +531,32 @@ def _push_message(self, payload, **kwargs):
                     _LOGGER.error("Error saving registration")
                 else:
                     _LOGGER.info("Configuration saved")
+
+
+def add_jwt(timestamp, target, tag, jwt_secret):
+    """Create JWT json to put into payload."""
+    import jwt
+    jwt_exp = (datetime.fromtimestamp(timestamp) +
+               timedelta(days=JWT_VALID_DAYS))
+    jwt_claims = {'exp': jwt_exp, 'nbf': timestamp,
+                  'iat': timestamp, ATTR_TARGET: target,
+                  ATTR_TAG: tag}
+    return jwt.encode(jwt_claims, jwt_secret).decode('utf-8')
+
+
+def create_vapid_headers(vapid_email, subscription_info, vapid_private_key):
+    """Create encrypted headers to send to WebPusher."""
+    from py_vapid import Vapid
+    try:
+        from urllib.parse import urlparse
+    except ImportError:  # pragma nocover
+        from urlparse import urlparse
+    if vapid_email and vapid_private_key:
+        url = urlparse(subscription_info.get('endpoint'))
```

Comment:
What if `endpoint` isn't there?

Reply:
if `endpoint` isn't there Webpusher will raise error before it.
https://github.com/web-push-libs/pywebpush/blob/2a23f45b7819e31bd030de9fe1357a1cf7dcfdc4/pywebpush/__init__.py#L139-L140

Should I raise it to?

Reply:
Yes

Reply:
Correction: You shouldn't raise it, you should log a descriptive error saying what went wrong.